pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/internal/templates/controllers/controller.go

```diff
@@ -0,0 +1,254 @@
+// pattern(https://kubernetes.io/docs/concepts/extend-kubernetes/operator/) you will create
+// Controllers(https://kubernetes.io/docs/concepts/architecture/controller/) which provide a reconcile function
+// responsible for synchronizing resources until the desired state is reached on the cluster. Breaking this
+// recommendation goes against the design principles of Controller-runtime(https://github.com/kubernetes-sigs/controller-runtime) 
+// and may lead to unforeseen consequences such as resources becoming stuck and requiring manual intervention.
+//
+// For more details, check Reconcile and its Result here:
+// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@{{ .ControllerRuntimeVersion }}/pkg/reconcile
+func (r *{{ .Resource.Kind }}Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := ctrllog.FromContext(ctx)
+
+	// Fetch the {{ .Resource.Kind }} instance
+	// The purpose is check if the Custom Resource for the Kind {{ .Resource.Kind }}
+	// is applied on the cluster if not we return nill to stop the reconciliation
+	{{ .Resource.Kind }} := &{{ .Resource.ImportAlias }}.{{ .Resource.Kind }}{}
+	err := r.Get(ctx, req.NamespacedName, {{ .Resource.Kind }})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			log.Info("{{ .Resource.Kind }} resource not found. Ignoring since object must be deleted")
+			return ctrl.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		log.Error(err, "Failed to get .Resource.Kind }}")
+		return ctrl.Result{}, err
+	}
+
+	// Check if the deployment already exists, if not create a new one
+	found := &appsv1.Deployment{}
+	err = r.Get(ctx, types.NamespacedName{Name: {{ .Resource.Kind }}.Name, Namespace: {{ .Resource.Kind }}.Namespace}, found)
+	if err != nil && errors.IsNotFound(err) {
+		// Define a new deployment
+		dep := r.deploymentFor{{ .Resource.Kind }}({{ .Resource.Kind }})
+		log.Info("Creating a new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+		err = r.Create(ctx, dep)
+		if err != nil {
+			log.Error(err, "Failed to create new Deployment", "Deployment.Namespace", dep.Namespace, "Deployment.Name", dep.Name)
+			return ctrl.Result{}, err
+		}
+		// Deployment created successfully 
+		// We will requeue the reconciliation so that we can ensure the state
+		// and move forward for the next operations
+		return ctrl.Result{Requeue: true}, nil
+	} else if err != nil {
+		log.Error(err, "Failed to get Deployment")
+		// Let's return the error for the reconciliation be re-trigged again 
+		return ctrl.Result{}, err
+	}
+
+	// The API is defining that the {{ .Resource.Kind }} type, have a {{ Resource.Kind }}Spec.Size field to set the quantity of {{ Resource.Kind }} instances (CRs) to be deployed. 
```

Comment:
@camilamacedo86 By running create api command this statement is giving error:
```
failed to create API: unable to scaffold with "deploy-image.go.kubebuilder.io/v1-alpha": error updating scaffold: template: *controllers.Controller:93: function "Resource" not defined
```

Reply:
That means we have errors in the boilerplate.